tests/index.test.ts

```diff
@@ -127,6 +127,29 @@ describe(
       expect(getOpenFds()).toStrictEqual(oldFds);
     });
 
+    test("cannot be written to after closing", async () => {
+      const onExit = vi.fn();
+      const pty = new Pty({
+        command: '/bin/echo',
+        args: ['hello'],
+        onExit,
+      });
+
+      const writeStream = pty.write;
+
+      const errorPromise = new Promise<void>((resolve, reject) => {
+        writeStream.on('error', (error) => {
+          reject(error);
+        });
+      });
+
+      pty.close();
+
+      expect(writeStream.writable).toBe(false);
+      writeStream.write("hello");
+      await expect(errorPromise).rejects.toThrowError(/write after end/);
+    });
```

Comment:
I think I don't understand the context enough but why does this change fix the increased `Cannot call write after a stream was destroyed` errors in pid2, this test seems to suggest it will just convert them to `write after end` errors instead? Maybe these are handled differently somewhere?